sql/handler.cc

```diff
@@ -7369,11 +7360,16 @@ bool Vers_parse_info::fix_alter_info(THD *thd, Alter_info *alter_info,
           return true;
         }
 
-        if (vers_change_sys_field(thd, name, alter_info,
-                                  f->flags & VERS_SYSTEM_FIELD, name))
-        {
+        bool integer= table->vers_start_field()->type() == MYSQL_TYPE_LONGLONG;
+        Create_field *field= vers_init_sys_field(thd, name, f->flags & VERS_SYSTEM_FIELD, integer);
+        if (!field)
           return true;
-        }
+
+        field->change.str= name;
+        field->change.length= strlen(name);
```

Comment:
Did you check that `strlen()` of **name** is known?

Reply:
That can be improved. Thanks for noticing.

Reply:
Fixed.